### PR TITLE
1131 add policy for updating course

### PR DESCRIPTION
**Pull request by defong**
### Context
policy for updating course

### Changes proposed in this pull request
Added policy for updating course

### Guidance to review
As long as the user has access to the course, for example to view it, then inherently they can update it as well.

### CODE REVIEW

**Comment by defong** [comment]
[API endpoint for updating vacancies](https://trello.com/c/UnhePmKO/1131-api-endpoint-for-updating-vacancies)

File: spec/policies/course_policy_spec.rb [hunk 1]
```diff
@@ -35,6 +39,10 @@
       it 'allows the :show action' do
         expect(subject.show?).to be_falsey
       end
+
+      it 'allows the :update action' do
```
Comment (misaka) [inline spec/policies/course_policy_spec.rb]: The words aren't quite right here.

File: spec/policies/course_policy_spec.rb [hunk 2]
```diff
@@ -27,6 +27,10 @@
       it 'allows the :show action' do
         expect(subject.show?).to be_truthy
       end
+
+      it 'allows the :update action' do
+        expect(subject.update?).to be_truthy
+      end
     end
 
     context 'when accessing a course of a provider that the user does not have' do
```
Comment (misaka) [inline spec/policies/course_policy_spec.rb]: Just realised, there are custom matchers to test policies:

https://github.com/varvet/pundit#policy-specs

File: spec/policies/course_policy_spec.rb (hunk 1 above)
Comment (defong) [reply spec/policies/course_policy_spec.rb]: allows the :show action
Vs 
allows the :update action
Vs
... 

Not sure what's you mean...

Comment (defong) [reply spec/policies/course_policy_spec.rb]: https://github.com/varvet/pundit/blob/master/spec/policies/post_policy_spec.rb


Actually the custom mishy matcher will get rid of my bad English

Comment (defong) [reply spec/policies/course_policy_spec.rb]: Guess I need to look at this as well.

https://github.com/varvet/pundit/blob/master/spec/spec_helper.rb

Cheers misha

**Pull request merged**